controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -563,20 +534,14 @@ func (r *ROSAControlPlaneReconciler) reconcileClusterAdminPassword(ctx context.C
 	return password, nil
 }
 
-func validateControlPlaneSpec(rosaClient *rosa.RosaClient, rosaScope *scope.ROSAControlPlaneScope) (*string, error) {
+func validateControlPlaneSpec(ocmClient *ocm.Client, rosaScope *scope.ROSAControlPlaneScope) error {
 	version := rosaScope.ControlPlane.Spec.Version
-	isSupported, err := rosaClient.IsVersionSupported(version)
-	if err != nil {
-		return nil, fmt.Errorf("failed to verify if version is supported: %w", err)
-	}
-
-	if !isSupported {
-		message := fmt.Sprintf("version %s is not supported", version)
-		return &message, nil
+	if _, err := ocmClient.ValidateHypershiftVersion(version, "stable"); err != nil {
```

Comment:
why are we ignoring the return value of `ValidateHypershiftVersion()`?
This function return a bool if the version is valid, an error is only returned if there was an error during the API call.

Also why are we changing the `validateControlPlaneSpec()` func return signature? we need to distinguish between user invalid input and real errors to inform whether to requeue or not, API errors for example could be intermittent and be resolved when we requeue.